hack/loadtest/util/placement.go

```diff
@@ -203,24 +229,77 @@ func collectDeleteMetrics(ctx context.Context, hubClient client.Client, deadline
 		}
 		if allRemoved {
 			// succeeded
-			klog.V(3).Infof("the cluster resource placement `%s` delete succeeded", crpName)
+			klog.V(3).Infof("the applied resources on cluster resource placement `%s` delete succeeded", crpName)
 			LoadTestDeleteCountMetric.WithLabelValues(currency, fleetSize, "succeed").Inc()
 			deleteSuccessCount.Inc()
 			LoadTestDeleteLatencyMetric.WithLabelValues(currency, fleetSize).Observe(time.Since(startTime).Seconds())
 			break
 		}
 		if time.Now().After(deleteDeadline) {
 			// timeout
-			klog.V(3).Infof("the cluster resource placement `%s` delete timeout", crpName)
+			klog.V(3).Infof("the applied resources on cluster resource placement `%s` delete timeout", crpName)
 			LoadTestDeleteCountMetric.WithLabelValues(currency, fleetSize, "timeout").Inc()
 			deleteTimeoutCount.Inc()
 			break
 		}
 	}
 }
 
+// check crp updated/completed before deletion
+func waitForCrpToComplete(ctx context.Context, hubClient client.Client, deadline, pollInterval time.Duration, deletionStartTime time.Time, crpName string, currency string, fleetSize string) {
+	startTime := time.Now()
+	applyDeadline := startTime.Add(deadline)
+	var crp v1beta1.ClusterResourcePlacement
+	var err error
+	for ; ctx.Err() == nil; time.Sleep(pollInterval) {
+		if err = hubClient.Get(ctx, types.NamespacedName{Name: crpName, Namespace: ""}, &crp); err != nil {
+			klog.ErrorS(err, "failed to get crp", "crp", crpName)
+			if time.Now().After(applyDeadline) {
+				// timeout
+				klog.V(2).Infof("the cluster resource placement `%s` timeout", crpName)
+				LoadTestUpdateCountMetric.WithLabelValues(currency, fleetSize, "timeout").Inc()
+				updateTimeoutCount.Inc()
+				break
+			}
+			continue
+		}
+		// check if the condition is true
+		appliedCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementAppliedConditionType))
+		synchronizedCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementSynchronizedConditionType))
+		scheduledCond := crp.GetCondition(string(v1beta1.ClusterResourcePlacementScheduledConditionType))
+		if appliedCond == nil || appliedCond.Status == metav1.ConditionUnknown || synchronizedCond == nil || synchronizedCond.Status == metav1.ConditionUnknown || scheduledCond == nil || scheduledCond.Status == metav1.ConditionUnknown {
```

Comment:
Hi Britania! There is a `IsConditionTrue` utility in the conditon package (from Fleet's own repo), that might be of use here.